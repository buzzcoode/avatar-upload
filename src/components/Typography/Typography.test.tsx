import React from 'react';
import { render } from '@testing-library/react';
import { Typography } from '.';

it('renders correctly', async () => {
  const { queryByTestId } = render(<Typography>content</Typography>);
  expect(queryByTestId('styled-typography')).toMatchSnapshot();
});

it('renders title variant snapshot', async () => {
  const { container } = render(<Typography variant="title">title typography</Typography>);
  expect(container).toMatchSnapshot();
});

it('renders text variant snapshot', async () => {
  const { container } = render(<Typography variant="text">text typography</Typography>);
  expect(container).toMatchSnapshot();
});

it('renders error variant snapshot', async () => {
  const { container } = render(<Typography variant="error">error typography</Typography>);
  expect(container).toMatchSnapshot();
});

it('renders default variant snapshot', async () => {
  const { container } = render(<Typography>typography</Typography>);
  expect(container).toMatchSnapshot();
});
