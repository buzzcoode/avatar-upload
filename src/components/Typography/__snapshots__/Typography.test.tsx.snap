// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  class="css-srpoh3"
  data-testid="styled-typography"
>
  content
</span>
`;

exports[`renders default variant snapshot 1`] = `
<div>
  <span
    class="emotion-0"
    data-testid="styled-typography"
  >
    typography
  </span>
</div>
`;

exports[`renders error variant snapshot 1`] = `
.emotion-0 {
  font-weight: 400;
  font-size: 16px;
  line-height: 28.8px;
  color: error;
}

<div>
  <span
    class="emotion-0"
    data-testid="styled-typography"
    font-size="16px"
    font-weight="400"
  >
    error typography
  </span>
</div>
`;

exports[`renders text variant snapshot 1`] = `
.emotion-0 {
  font-weight: 400;
  font-size: 14px;
  line-height: 25.2px;
}

<div>
  <span
    class="emotion-0"
    data-testid="styled-typography"
    font-size="14px"
    font-weight="400"
  >
    text typography
  </span>
</div>
`;

exports[`renders title variant snapshot 1`] = `
.emotion-0 {
  font-weight: 500;
  font-size: 14px;
  line-height: 25.2px;
}

<div>
  <span
    class="emotion-0"
    data-testid="styled-typography"
    font-size="14px"
    font-weight="500"
  >
    title typography
  </span>
</div>
`;
